src/main/java/com/unascribed/fabrication/mixin/g_weird_tweaks/source_dependant_iframes/MixinLivingEntity.java

```diff
@@ -0,0 +1,62 @@
+package com.unascribed.fabrication.mixin.g_weird_tweaks.source_dependant_iframes;
+
+import com.unascribed.fabrication.interfaces.TickSourceIFrames;
+import com.unascribed.fabrication.support.EligibleIf;
+import com.unascribed.fabrication.support.MixinConfigPlugin;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.damage.DamageSource;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.world.World;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@Mixin(LivingEntity.class)
+@EligibleIf(configAvailable="*.source_dependant_iframes")
+public abstract class MixinLivingEntity extends Entity implements TickSourceIFrames {
+
+	public MixinLivingEntity(EntityType<?> type, World world) {
+		super(type, world);
+	}
+
+	Set<String> fabrication$iframeTracker = new HashSet<>();
```

Comment:
Should this not be a Map<String, Integer>? Having every source share the same iframe timer feels a little weird